### ISSUE TITLE
Retriever `run` method to accept `query`

### ISSUE THREAD

**Issue by TuanaCelik** [masci/chroma-haystack#6]
This is more of a question than it is an issue:
- It seems like in Haystack, we're moving towards having run methods accept `query` instead of `queries`, like [here in the BM25 Retriever for the memory store](https://github.com/deepset-ai/haystack/blob/35ec8cc8fb310d8a9b6ccea72ed74858d61d4d1a/haystack/preview/components/retrievers/memory_bm25_retriever.py#L74), if that's the case, would it make sense to change the retriever here to do the same?
- The embedding retriever for the memory store in the core project doesn't take `query` but `query_embedding`. The retriever here is called a dense retriever, should it also expect a query embedding as input?

**Comment by masci** [comment]
> * It seems like in Haystack, we're moving towards having run methods accept `query` instead of `queries`, like [here in the BM25 Retriever for the memory store](https://github.com/deepset-ai/haystack/blob/35ec8cc8fb310d8a9b6ccea72ed74858d61d4d1a/haystack/preview/components/retrievers/memory_bm25_retriever.py#L74), if that's the case, would it make sense to change the retriever here to do the same?

See https://github.com/masci/chroma-haystack/pull/7

**Comment by masci** [comment]
> * The embedding retriever for the memory store in the core project doesn't take `query` but `query_embedding`. The retriever here is called a dense retriever, should it also expect a query embedding as input?

See https://github.com/masci/chroma-haystack/pull/8 - I added another retriever that takes the embedding of a query as input and returns a list of documents

**Comment by masci** [comment]
@TuanaCelik let me know if the two PRs make sense, in case that's the case we can close the issue.

**Comment by TuanaCelik** [comment]
Makes sense to me @masci - I guess we could rethink the naming at some point, for example 'QueryRetriever' makes it sound like it's going to retrieve queries, not retrieve 'documents' _via_ query :)
But I don't have a good alternative to suggest for now. I will send this Issue and PRs to MarqoDB so that they also see what we've changed in the DocumentStore API.
